shell/packages/sandstorm-ui-powerbox/powerbox-server.js

```diff
@@ -41,28 +44,61 @@ Meteor.methods({
 
     const db = this.connection.sandstormDb;
 
+    const session = db.collections.sessions.findOne(sessionId);
+    if (!session) {
+      throw new Meteor.Error(403, "Invalid session ID");
+    }
+
     let descriptor;
+    let requirements = [];
     if (frontendRefVariety.ipNetwork) {
       if (!db.isAdmin(this.userId)) {
         throw new Meteor.Error(403, "User must be an admin to powerbox offer IpNetwork");
       }
 
       descriptor = encodePowerboxDescriptor({ tags: [{ id: Ip.IpNetwork.typeId }] });
+      requirements.push({ userIsAdmin: Meteor.userId() });
     } else if (frontendRefVariety.ipInterface) {
       if (!db.isAdmin(this.userId)) {
         throw new Meteor.Error(403, "User must be an admin to powerbox offer IpInterface");
       }
 
       descriptor = encodePowerboxDescriptor({ tags: [{ id: Ip.IpInterface.typeId }] });
+      requirements.push({ userIsAdmin: Meteor.userId() });
```

Comment:
As a matter of practicality, I'm not sure you want this requirement, given the intended use case of IpNetwork and IpInterface for drivers:

1. Most of the time, admins won't have their admin privileges revoked.
2. We have UI in the admin panel for admins to review and revoke all IpNetwork and IpInterface sturydyrefs.
3. If an admin leaves the company or changes roles and no longer needs admin privileges, this requirement necessitates urgently reissuing the capability to any drivers they may have granted the privilege to.
4. I assume that most users will not actually feel a need to reissue the capabilities that that admin initially set up.
5. We currently lack a system for methodically reissuing such capabilities, and building an automated one will require explicit support from apps/drivers.

I suspect that in practice, having IpNetwork/IpInterface auto-revoke when the admin becomes a non-admin is likely to be **much** more surprising and damaging to real deployments than letting them persist.

The way I'd been thinking about things, IpNetwork and IpInterface are owned by the deployment, not the admin that granted them, and admins have an interface by which they may ask the deployment to create/modify/delete them.  Maybe that's not quite the pure ocap model, but it is how I'd been modeling the problem space, especially given that admins have/need the power to review/revoke each other's IpNetwork/IpInterface grants.